cpp/include/cuspatial/distances/point_distance.hpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/utilities/span.hpp>
+
+namespace cuspatial {
+
+/**
+ * @ingroup distance
+ * @brief Compute pairwise point to point cartesian distance
+ *
+ * @param points1_x Column of x coordinates to the first point in each pair
+ * @param points1_y Column of y coordinates to the first point in each pair
+ * @param points2_x Column of x coordinates to the second point in each pair
+ * @param points2_y Column of y coordinates to the second point in each pair
+ * @param stream The CUDA stream to use for device memory operations and kernel launches
+ * @return Column of distances between each pair of input points
+ */
+std::unique_ptr<cudf::column> pairwise_point_distance(
```

Comment:
For discussion: thinking forward about naming: 

point-point distance: `pairwise_point_distance`
linestring-linestring distance; `pairwise_linestring_distance`
point-linestring distance: `pairwise_point_linestring_distance` (or is it `pairwise_linestring_point_distance`?)

Should we standardize on the number of geometries in the name of the function? E.g. should point-point distance be called `pairwise_point_point_distance`? Or is it obvious that `point` means `point-to-point`?  I kind of think it is, and it makes for shorter names.

Second, how do we order the geometries? Should the more complex geometry be first, or second?

What if there is a directed distance metric (e.g. Hausdorff is directed -- asymmetric). Do we need to support both point-linestring and linestring-point? I think we do need it one way or another...

Reply:
I looked at libgeos, geopandas and postgis and none give good hint about naming. So I say my feelings here. Personally I think `pairwise_point_distance` is sufficient to represent `point-to-point distance`. Because of the notion `pair`, it's unambiguous that two geometries are involved.

> Second, how do we order the geometries? Should the more complex geometry be first, or second?

This is what I felt too. But in more explicit way is to sort the geometries by dimensions.

> What if there is a directed distance metric 

One way is to use an extra enum argument `reflected`. The benefits are we don't have to break the ordering rule above while still providing the capability. The downside is that for same geometry (linestring-linestring), it's arguably redundant.

I also want to note the grouping of functions. The `pairwise_point_distance` should live under `distance/point_distance.hpp`. The name of the method "XXX_point_distance" should match the name of the file. In case in the future we need to implement a different parallel model e.g. `all_pairs_point_distance`, it should also live under this file. 

If you agree, I can make an issue about the names of distance method.

Reply:
https://github.com/rapidsai/cuspatial/issues/607